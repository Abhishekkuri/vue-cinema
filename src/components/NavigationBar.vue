<template>
  <div class="navigation-container">
    <nav>
      <router-link class="nav-link" to="/">Featured Films</router-link>
      <input
        type="text"
        v-model="title"
        placeholder="Search for a Movie"
        class="nav-input"
      />
      <button class="nav-button" @click="searchMovies">Search</button>
      <p class="error-message" v-if="searchedMovies.length === 0">
        Oops! try searching film with some other title.
      </p>
    </nav>
  </div>
</template>

<script>
import { mapGetters } from "vuex";
export default {
  data() {
    return {
      title: "",
    };
  },
  computed: {
    ...mapGetters({ searchedMovies: "getSearchedFilms" }),
  },
  methods: {
    async searchMovies() {
      this.showErrorMessage = false;
      await this.$store.dispatch("fetchSearchedFilms", {
        title: this.title,
      });
      this.$router.replace(`/search/${this.title}`);
    },
  },
};
</script>

<style lang="scss" scoped>
.navigation-container {
  background-color: $color-grey-light;
  nav {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    font-size: 1.8rem;
    padding: 1rem;
    margin: 1rem;
    .nav-link {
      color: $color-black;
      border: 1px solid $color-white;
      text-decoration: none;
      padding: 1rem;
      margin-bottom: 4rem;
      border-radius: 10px;
      transition: 0.3s;
      &:hover {
        border: none;
        border: 1px solid $color-black;
        color: $color-grey-light;
        background-color: $color-primary;
        box-shadow: 0 8px 16px 0 $color-primary;
      }
    }
    .router-link-exact-active {
      color: $color-tirtiery;
      border-bottom: 1px solid $color-primary;
      border-radius: 0;
      border-top: none;
      border-left: 0;
      border-right: 0;
    }
    .nav-input {
      padding: 1rem;
      font-size: 1.5rem;
      border: 1px solid $color-tirtiery;
      border-radius: 10px;
      text-align: center;
      transition: 0.3s;
      &:hover {
        border: 1px solid $color-black;
        color: $color-grey-light;
        background-color: $color-primary;
        box-shadow: 0 8px 16px 0 $color-primary;
      }
    }
    .nav-button {
      padding: 1rem 2rem;
      font-size: 1.5rem;
      cursor: pointer;
      background-color: $color-grey-light;
      border: none;
      border-radius: 15px;
      margin-top: 1.5rem;
      transition: 0.3s;
      &:hover {
        border: 1px solid $color-black;
        color: $color-grey-light;
        background-color: $color-primary;
        box-shadow: 0 8px 16px 0 $color-primary;
      }
    }
  }
}

.error-message {
  color: red;
  font-size: 1rem;
  text-align: center;
}

@media screen and (max-width: 820px) {
  .navigation-container {
    nav {
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-evenly;
      .nav-link {
        font-size: 1rem;
        padding: 1rem;
        margin: 0.5rem;
      }
      .nav-input {
        font-size: 1rem;
        padding: 1rem;
        margin: 0.5rem;
      }
      .nav-button {
        font-size: 1rem;
        padding: 0.5rem;
        margin: 0.5rem;
      }
    }
  }
}
</style>
