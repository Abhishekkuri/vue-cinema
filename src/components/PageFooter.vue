<template>
  <div class="footer-container">
    <span>This is a Cinema application.</span>
    <span>Framework: VueJS</span>
    <span>Creator: Abhishek Kuri</span>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.footer-container {
  background-color: $color-tirtiery;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
}
</style>
