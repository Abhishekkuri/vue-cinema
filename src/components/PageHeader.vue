<template>
  <div class="header-container">
    <img src="@/assets/cinema-trans.png" alt="logo" />
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.header-container {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: $color-primary;
  img {
    height: 20rem;
    width: 20rem;
  }
}
</style>
