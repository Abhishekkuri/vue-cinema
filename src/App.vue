<template>
  <div class="main-container">
    <page-header class="page-header"></page-header>
    <navigation-bar class="navigation-bar"> </navigation-bar>
    <transition name="fade" mode="out-in">
      <router-view class="router-view" />
    </transition>
    <page-footer class="page-footer" />
  </div>
</template>

<script>
import NavigationBar from "@/components/NavigationBar.vue";
import PageFooter from "@/components/PageFooter.vue";
import PageHeader from "@/components/PageHeader.vue";

export default {
  components: {
    PageHeader,
    PageFooter,
    NavigationBar,
  },
};
</script>

<style lang="scss">
html,
body {
  box-sizing: border-box;
  margin: 0px !important;
  padding: 0 !important;
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.view-heading {
  display: block;
  text-align: center;
  font-size: 3rem;
  color: $color-primary;
  padding-top: 1rem;
  text-transform: uppercase;
}

.fade-enter-active,
.fade-leave-active {
  transition-duration: 0.4s;
  transition-property: opacity;
  transition-timing-function: ease;
}

.fade-enter,
.fade-leave-active {
  opacity: 0;
}

.page-loader {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
}
</style>

<style lang="scss" scoped>
.main-container {
  width: 100vw;
  min-height: 100vh;
  background-color: $color-white;
  display: grid;
  grid-template-columns: 1.5fr 4fr;
  grid-template-rows: 10rem 4fr 5rem;
  .page-header {
    grid-column: 1/3;
  }
  .navigation-bar {
    grid-column: 1/2;
  }
  .router-view {
    grid-column: 2/3;
  }
  .page-footer {
    grid-column: 1/3;
  }
}

@media screen and (max-width: 820px) {
  .view-heading {
    font-size: 2rem;
    padding-top: 0.5rem;
  }

  .main-container {
    width: 100vw;
    min-height: 100vh;
    background-color: $color-white;
    display: grid;
    grid-template-rows: 10rem 10rem 4fr 5rem;
  }
  .page-header {
    grid-row: 1/2;
    grid-column: 1/3 !important;
  }
  .navigation-bar {
    grid-row: 2/3;
    grid-column: 1/3 !important;
  }
  .router-view {
    grid-row: 3/4;
    grid-column: 1/3 !important;
  }
  .page-footer {
    grid-row: 4/5;
    grid-column: 1/3 !important;
  }
}
</style>
