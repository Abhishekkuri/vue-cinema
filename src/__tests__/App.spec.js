import { shallowMount } from "@vue/test-utils";
import NavigationBar from "@/components/NavigationBar.vue";
import PageFooter from "@/components/PageFooter.vue";
import PageHeader from "@/components/PageHeader.vue";
import App from "@/App";

describe("FeaturedPage.vue", () => {
  test("Featured Page renders correctly", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.element).toMatchSnapshot();
  });
  test("Featured Page renders page header", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.findComponent(PageHeader).exists()).toBe(true);
  });
  test("Featured Page renders page footer", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.findComponent(PageFooter).exists()).toBe(true);
  });
  test("Featured Page renders navigation bar", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.findComponent(NavigationBar).exists()).toBe(true);
  });
});
