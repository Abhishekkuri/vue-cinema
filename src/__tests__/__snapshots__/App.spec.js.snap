// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedPage.vue Featured Page renders correctly 1`] = `
<div
  class="main-container"
>
  <page-header-stub
    class="page-header"
  />
   
  <navigation-bar-stub
    class="navigation-bar"
  />
   
  <transition-stub
    mode="out-in"
    name="fade"
  >
    <router-view
      class="router-view"
    />
  </transition-stub>
   
  <page-footer-stub
    class="page-footer"
  />
</div>
`;
